### PR TITLE
Set rollup initial operator set through aggregator

### PR DESCRIPTION
**Pull request by Hyodar**

### CODE REVIEW

File: aggregator/rollup_broadcaster.go [hunk 1]
```diff
@@ -49,15 +55,111 @@ func NewRollupWriter(ctx context.Context, rollupInfo config.RollupInfo, signerCo
...
+
+	if nextOperatorUpdateId == 0 {
+		go writer.initializeOperatorSet(ctx)
+	}
+
+	return writer, nil
+}
+
+func (w *RollupWriter) initializeOperatorSet(ctx context.Context) {
+	mainnetNextOperatorSetUpdateId, err := w.avsReader.GetNextOperatorSetUpdateId(ctx)
```
[the remark this answers is not in the document]
Comment (Hyodar) [reply aggregator/rollup_broadcaster.go]: Now using this only on brodcaster as per [493f586](https://github.com/NethermindEth/near-sffl/pull/68/commits/493f586259159fde4e08e8090265c20acfeb48d2). I still think it's relevant to differentiate it from the rollup ones.

File: aggregator/rollup_broadcaster.go [hunk 2]
```diff
@@ -49,15 +55,111 @@ func NewRollupWriter(ctx context.Context, rollupInfo config.RollupInfo, signerCo
...
+	if err != nil {
+		w.logger.Error("Error fetching operator set update id", "err", err)
+		return
+	}
+
+	var operatorSetUpdatedChan chan *opsetupdatereg.ContractSFFLOperatorSetUpdateRegistryOperatorSetUpdatedAtBlock
```
[the remark this answers is not in the document]
Comment (Hyodar) [reply aggregator/rollup_broadcaster.go]: Done in [493f586](https://github.com/NethermindEth/near-sffl/pull/68/commits/493f586259159fde4e08e8090265c20acfeb48d2)

File: aggregator/rollup_broadcaster.go [hunk 3]
```diff
@@ -49,15 +55,111 @@ func NewRollupWriter(ctx context.Context, rollupInfo config.RollupInfo, signerCo
 		return nil, err
 	}
 
-	return &RollupWriter{
+	writer := &RollupWriter{
 		txMgr:              txMgr,
 		client:             client,
 		sfflRegistryRollup: sfflRegistryRollup,
+		avsReader:          avsReader,
```
[the remark this answers is not in the document]
Comment (Hyodar) [reply aggregator/rollup_broadcaster.go]: Done in [493f586](https://github.com/NethermindEth/near-sffl/pull/68/commits/493f586259159fde4e08e8090265c20acfeb48d2)

File: aggregator/rollup_broadcaster.go [hunk 4]
```diff
@@ -49,15 +55,111 @@ func NewRollupWriter(ctx context.Context, rollupInfo config.RollupInfo, signerCo
...
+	w.operatorSetUpdateLock.Lock()
+	defer w.operatorSetUpdateLock.Unlock()
+
+	mainnetNextOperatorSetUpdateId, err := w.avsReader.GetNextOperatorSetUpdateId(ctx)
+	if err != nil {
+		w.logger.Error("Error fetching operator set update id", "err", err)
+		return
+	}
+
+	operators, err := w.avsReader.GetOperatorSetUpdateDelta(ctx, mainnetNextOperatorSetUpdateId-1)
```
[the remark this answers is not in the document]
Comment (Hyodar) [reply aggregator/rollup_broadcaster.go]: Added a utility in [15b5432](https://github.com/NethermindEth/near-sffl/pull/68/commits/15b5432171f75e2aedfc9c81eda8c13db255a54b) and used in [493f586](https://github.com/NethermindEth/near-sffl/pull/68/commits/493f586259159fde4e08e8090265c20acfeb48d2)

File: aggregator/rollup_broadcaster.go [hunk 5]
```diff
@@ -138,10 +221,99 @@ func NewRollupBroadcaster(ctx context.Context, rollupsInfo map[uint32]config.Rol
...
+
+func (b *RollupBroadcaster) tryInitializeRollupOperatorSet(ctx context.Context, writer *RollupWriter, mainnetNextOperatorSetUpdateId uint64, avsReader chainio.AvsReaderer, avsSubscriber chainio.AvsSubscriberer) {
+	nextOperatorUpdateId, err := writer.sfflRegistryRollup.NextOperatorUpdateId(&bind.CallOpts{})
+	if err != nil {
+		b.logger.Error("Error fetching NextOperatorUpdateId", "err", err)
+		b.errorChan <- err
+		return
+	}
+
+	if nextOperatorUpdateId == 0 {
```
Comment (taco-paco) [inline aggregator/rollup_broadcaster.go:293]: Minor, but I would propose to make it even better with early return statement.

```
	if nextOperatorUpdateId != 0 {
             return
	}

       // logic
```

**Review by taco-paco — APPROVED** [review]
Gorgeous! As final suggestion I think RollupBroadcaster & RollupWriter are matured enough to have their own files

File: aggregator/rollup_broadcaster.go [hunk 6]
```diff
@@ -164,3 +336,26 @@ func (b *RollupBroadcaster) BroadcastOperatorSetUpdate(ctx context.Context, mess
 func (b *RollupBroadcaster) GetErrorChan() <-chan error {
 	return b.errorChan
 }
+
+func (b *RollupBroadcaster) tryGetOperatorSetUpdateById(ctx context.Context, avsReader chainio.AvsReaderer, operatorSetUpdateId uint64) ([]opsetupdatereg.RollupOperatorsOperator, error) {
```
Comment (Hyodar) [inline aggregator/rollup_broadcaster.go:340]: Need to rename this one to `tryGetOperatorSetUpdateById`

File: aggregator/rollup_broadcaster.go (hunk 5 above)
Comment (Hyodar) [reply aggregator/rollup_broadcaster.go]: Agreed, done in [b06b38d](https://github.com/NethermindEth/near-sffl/pull/68/commits/b06b38dd2c6e3f5336fac1f80a4ea55ebaa7e61d)

File: aggregator/rollup_broadcaster.go (hunk 6 above)
Comment (Hyodar) [reply aggregator/rollup_broadcaster.go]: Done in [cf14930](https://github.com/NethermindEth/near-sffl/pull/68/commits/cf14930a988b41877ac3c6e7a45d8310554a60fe)

**Pull request merged**